procurement_mto_analytic/models/purchase_order_line.py

```diff
@@ -0,0 +1,24 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    def _merge_in_existing_line(
+            self, product_id, product_qty, product_uom, location_id, name,
+            origin, values):
+        # Do not merge into a line with a different analytic account
+
+        res = super(PurchaseOrderLine, self)._merge_in_existing_line
+
+        if len(self.account_analytic_id) == 0:
```

Comment:
@n17 Use like this `if not self.account_analytic_id:`, not checked len of account